### PR TITLE
Avoid to create an API client if not needed

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes https://github.com/openshift/origin/issues/9992

Only create an OpenShift API client if we are actually dealing with OpenShift objects. Will avoid trying to reach `/oapi` if we are only dealing with pods, which makes `oc logs` to keep compatibility with a raw Kube server.

@liggitt PTAL

### CODE REVIEW

**Comment by liggitt** [comment]
did you sweep other factory methods to make sure we're not doing this elsewhere?

**Comment by liggitt** [comment]
looks like several other factory methods have the same issue

**Comment by fabianofranz** [comment]
@liggitt yeah there might be. Checking.

**Comment by fabianofranz** [comment]
[test]

**Comment by fabianofranz** [comment]
@liggitt thanks for catching, there were a few others. Mind taking a look?

File: pkg/cmd/util/clientcmd/factory.go [hunk 1]
```diff
@@ -714,6 +731,10 @@ func (f *Factory) PodForResource(resource string, timeout time.Duration) (string
 	case api.Resource("pods"):
 		return name, nil
 	case api.Resource("replicationcontrollers"):
```
Comment (liggitt) [inline pkg/cmd/util/clientcmd/factory.go]: we could hit all these k8s resources and fail against a kubernetes server, right?

Comment (fabianofranz) [inline pkg/cmd/util/clientcmd/factory.go]: Right. I must use `f.Client()` instead of `f.Clients()` so it only takes the kube client. Fixed.

**Comment by liggitt** [comment]
LGTM

**Comment by fabianofranz** [comment]
[merge]

**Pull request merged**